adagucserverEC/Types/LayerMetadataType.h

```diff
@@ -0,0 +1,66 @@
+
+#ifndef LAYER_METADATA_TYPE_H
+#define LAYER_METADATA_TYPE_H
+
+#include "CTString.h"
+#include <CDataSource.h>
+
+struct LayerMetadataDim {
+  CT::string name;
+  CT::string units;
+  CT::string values;
+  CT::string defaultValue;
+  int hasMultipleValues;
+  bool hidden = false;
+};
+
+struct LayerMetadataProjection {
+  CT::string name;
+  double dfBBOX[4];
+};
+
+struct LayerMetadataStyle {
+  CT::string name;
+  CT::string title;
+  CT::string abstract;
+};
+
+struct LayerMetadataVariable {
+  CT::string units;
+};
+
+struct LayerMetadata {
+
+  int width = -1;
+  int height = -1;
+  double cellsizeX = 0;
+  double cellsizeY = 0;
+  double dfLatLonBBOX[4] = {-180, -90, 180, 90};
+  double dfBBOX[4] = {-180, -90, 180, 90};
+  int isQueryable = 0;
+  CT::string name, title, group, abstract, nativeEPSG;
+
+  std::vector<LayerMetadataProjection *> projectionList;
+  std::vector<LayerMetadataDim *> dimList;
+  std::vector<LayerMetadataStyle *> styleList;
+  std::vector<LayerMetadataVariable *> variableList;
```

Comment:
Why use pointers here, instead of putting the structs directly into the vectors?
Construct the structs with something like 
`LayerMetadataStyle sList = {name, title, abstact}`
followed by `push_back`.

If you are worried about performance:
1. Don't be, it won't matter.
2. Copy elision and move semantics probably remove all the copies anyway.
3. If you are still worried, provide a move constructor on the structs and use `emplace_back()`. 

And example of point 3 (but you don't need to go there...):
```
#include <iostream>
#include <vector>

struct style {
  std::string name;
  std::string title;
  std::string abstract;
  
  style(std::string name, std::string title, std::string abstract)
    : name(std::move(name)), title(std::move(title)), abstract(std::move(abstract))
    {}
};

int main() {
    style s = {"test32", "test31", "test30"};
    v.push_back(s);

    v.emplace_back("test34", "test33", "test99");
    
    std::cout<<v[0].name<<std::endl;
    std::cout<<v[1].abstract<<std::endl;
}
```
According to the [spec](https://en.cppreference.com/w/cpp/container/vector/emplace_back) this avoids all copies.

Reply:
Changed from pointers to directly using structs